compiler/z/codegen/snippet/XPLINKCallDescriptorSnippet.cpp

```diff
@@ -208,11 +209,13 @@ TR::XPLINKCallDescriptorSnippet::XPLINKCallDescriptorSnippet(TR::CodeGenerator*
 uint8_t*
 TR::XPLINKCallDescriptorSnippet::emitSnippetBody()
    {
-   uint8_t* cursor = cg()->getBinaryBufferCursor();
-
+   uint8_t* cursor = reinterpret_cast<uint8_t*>(OMR::align(reinterpret_cast<size_t>(cg()->getBinaryBufferCursor()), 8));
```

Comment:
Do we need to update the length to account for the extra bytes needed to achieve doubleword alignment?

Reply:
Good point. I had not thought of this. I think it will screw up the estimation logic. There is a non-fatal assert for it. There does not appear to be an easy way to align general snippets and for that to play nice with the estimation because we are not guaranteed any alignment for the start of the literal pool.

However we do have guaranteed alignment for "data snippets" which are aligned to at least the size of the snippet. In our case our snippet is 8 bytes in length so it will be doubleword aligned. I have made the changes accordingly in 6810584 to make the `XPLINKCallDescriptorSnippet` inherit from `S390ConstantDataSnippet` which will be aligned (the fatal assert remains to ensure that).

Estimation will now work correctly as with all other constant data snippets. Good catch! Thanks for the review.

Reply:
I've also fixed the typos in the commit message of 826867e as per the above review.